### PR TITLE
[SPARK-14278][SQL] Initialize columnar batch with proper memory mode

### PR DESCRIPTION
**Pull request by sameeragarwal**
## What changes were proposed in this pull request?

Fixes a minor bug in the record reader constructor that was possibly introduced during refactoring. 
## How was this patch tested?

N/A

### CODE REVIEW

**Comment by sameeragarwal** [comment]
cc @marmbrus

**Comment by nongli** [comment]
LGTM

File: sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java [hunk 1]
```diff
@@ -191,7 +191,7 @@ public void initBatch(MemoryMode memMode, StructType partitionColumns,
       }
     }
 
-    columnarBatch = ColumnarBatch.allocate(batchSchema);
+    columnarBatch = ColumnarBatch.allocate(batchSchema, memMode);
```
Comment (JoshRosen) [inline sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java]: Can you avoid this class of problem in the future by simply removing the single-argument `allocate` method or the default value of `memMode`?

Comment (sameeragarwal) [inline sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java]: I considered it, but the single-argument `allocate` is currently also used in `DataSourceStrategy`. Can we change it there? cc @nongli

**Pull request closed without merging**